libraries/botbuilder/tests/teams/roster/src/rosterBot.ts

```diff
@@ -0,0 +1,132 @@
+        {
+            case "show members":
+                await this.showMembers(context);
+                break;
+
+            case "show channels":
+                await this.showChannels(context);
+                break;
+
+            case "show details":
+                await this.showDetails(context);
+                break;
+
+            default:
+                await context.sendActivity(this.createReply(context.activity, 
+                    'Invalid command. Type "Show channels" to see a channel list. Type "Show members" to see a list of members in a team. ' +
+                    'Type "show group chat members" to see members in a group chat.'));
+                break;
+        }
+    } 
+    
+    private createReply(activity, text = null, locale = null) : Activity {
+        return {
+            type: 'message',
+            from: { id: activity.recipient.id, name: activity.recipient.name },
+            recipient: { id: activity.from.id, name: activity.from.name },
+            replyToId: activity.id,
+            serviceUrl: activity.serviceUrl,
+            channelId: activity.channelId,
+            conversation: { isGroup: activity.conversation.isGroup, id: activity.conversation.id, name: activity.conversation.name },
+            text: text || '',
+            locale: locale || activity.locale
+        } as Activity;
+    }
+
+    private async showMembers(context: TurnContext): Promise<void> {
+        let teamsChannelAccounts = await this.getMembers(context);
+        let replyActivity = this.createReply(context.activity, `Total of ${teamsChannelAccounts.length} members are currently in team`);
+        await context.sendActivity(replyActivity);
+
+        var messages = teamsChannelAccounts.map(function(teamsChannelAccount){
+            return `${teamsChannelAccount.AadObjectId} --> ${teamsChannelAccount.Name} --> ${teamsChannelAccount.UserPrincipalName}`;
+        });
+
+        await this.sendInBatches(context, messages);
+    }
+    
+    private async showChannels(context: TurnContext): Promise<void> { 
+        let channels = await this.getChannels(context);
+        await context.sendActivity(this.createReply(context.activity, `Total of ${channels.length} channels are currently in team`));
+
+        var messages = channels.map(function(channel){
+            return `${channel.aadObjectId} --> ${channel.name} --> ${channel.userPrincipalName}`;
+        });
+
+        await this.sendInBatches(context, messages);
+    }
+   
+    private async showDetails(context: TurnContext): Promise<void> {
+        let teamDetails = await this.getTeamDetails(context);
+        var replyActivity = this.createReply(context.activity, `The team name is ${teamDetails.Name}. The team ID is ${teamDetails.Id}. The AAD GroupID is ${teamDetails.AadGroupId}.`);
+        await context.sendActivity(replyActivity);
+    }
+
+    private async getMembers(context: TurnContext): Promise<any[]> {
+        return null;
```

Comment:
>null [](start = 15, length = 4)

Assuming these are stubbed until in RosterClient completed..

Reply:
yup!